prac_07/project_management.py

```diff
@@ -0,0 +1,172 @@
+        for project in sorted_project_information:
+            entry_number += 1
+            formatted_date = project.start_date.strftime('%d/%m/%Y')
+            print(f"{entry_number} - {project.name}, start: {formatted_date}, priority = {project.priority}, "
+                  f"estimate: ${project.cost_estimate}, completion: {project.completion_percentage}%")
+    return sorted_project_information
+
+
+def read_from_file():
+    """Read project information from the file specified by the user"""
+    project_information = []
+    completed_projects = []
+    incomplete_projects = []
+    print("The name of the file is:")
+    file_name = input(">>>")
+    try:
+        in_file = open(file_name, 'r')
+        in_file.readline()
+        for line in in_file:
+            parts = line.strip().split('\t')
+            project_information.append(Project(parts[0], parts[1], parts[2], parts[3], parts[4]))
+    except FileNotFoundError:
+        print(f"{file_name} not found")
+    finally:
+        in_file.close()  # no problem with potential undefined variable
+    return project_information
+
+
+def save_to_file(project_list):
+    """save project information to the file specified by the user"""
+    print("The name of the file is:")
+    file_name = input(">>>")
+
+    try:
+        out_file = open(file_name, 'w')
+        print("Name\tStart Date\tPriority\tCost Estimate\tCompletion Percentage", file=out_file)
+        for line in project_list:
+            parts = [line.name, str(line.start_date), line.priority, line.cost_estimate, line.completion_percentage]
+            output = '\t'.join(parts)
+            print(output, file=out_file)
+    except FileNotFoundError:
+        print(f"{file_name} not found")
+    finally:
+        out_file.close()  # no problem with potential undefined variable
+
+
+def add_new_project():
+    """Get user input to add new project"""
+    print("Add new project:")
+    new_name = input("Project name: ")
+    new_start_date = input("Start date: ")
+    new_priority = input("Project priority: ")
+    new_cost_estimate = input("Project cost estimate: $")
+    new_completion_percentage = input("Project completion percentage: ")
+    new_project = Project(new_name, new_start_date, new_priority, new_cost_estimate, new_completion_percentage)
+    return new_project
+
+
+def update_project_information(project_information):
+    completed_projects = []
+    incomplete_projects = []
+    print("Enter the number of a project to modify its completion percentage and priority.")
+    project_number = int(input("Project Number: "))
+    print(f"{project_information[project_number - 1].name}, "
+          f"start: {project_information[project_number - 1].start_date}, "
+          f"priority = {project_information[project_number - 1].priority}, "
+          f"estimate: ${project_information[project_number - 1].cost_estimate}, "
+          f"completion: {project_information[project_number - 1].completion_percentage}%")
```

Comment:
project_number - 1 should be extracted to a variable as it is used multiple times. This shouldn't be necessary as the user should just be selecting the index itself, as per the sample output.